### ISSUE TITLE
Out of memory crash

### ISSUE THREAD

**Issue by alparp27** [DataDog/dd-sdk-android#164]
**Describe what happened**

```
Fatal Exception: java.lang.OutOfMemoryError
Failed to allocate a 32784 byte allocation with 23056 free bytes and 22KB until OOM, target footprint 268435456, growth limit 268435456

java.util.HashMap.resize + 703 (HashMap.java:703)
--
  | java.util.LinkedHashSet.<init> + 169 (LinkedHashSet.java:169)
  | kotlin.collections.CollectionsKt___CollectionsKt.toMutableSet + 1452 (_Collections.kt:1452)
  | kotlin.collections.CollectionsKt___CollectionsKt.union + 1467 (_Collections.kt:1467)
  | com.datadog.android.core.internal.data.file.FileReader.readNextFile + 70 (FileReader.kt:70)
  | com.datadog.android.core.internal.data.file.FileReader.readNextBatch + 29 (FileReader.kt:29)
  | com.datadog.android.core.internal.data.upload.DataUploadRunnable.run + 36 (DataUploadRunnable.kt:36)
  | android.os.Handler.handleCallback + 883 (Handler.java:883)
  | android.os.HandlerThread.run + 67 (HandlerThread.java:67)
```

**Steps to reproduce the issue:**
No known steps

**Describe what you expected:**
App would not crash at least

**Additional context**
13 instances of this so far and success rate is 99+% so this is hard to reproduce
samsung 100%

Operating Systems
10 54%
7 23%
9 23%

 - Datadog SDK version:1.2.1

**Comment by xgouchet** [comment]
Hi @alparp27 , thanks for submitting this issue.

That's indeed an issue that happens in extreme conditions: the device only has 23056 free bytes, that is less than 24K. We'll try to ensure that those crash don't happen in our library, although OOM errors are hard to recover from, and chances are it will only delay it.

**Comment by xgouchet** [comment]
This issue should be fixed as of `1.3.0+`
